executor/merge_join.go

```diff
@@ -109,51 +117,75 @@ func (t *mergeJoinInnerTable) init(ctx context.Context, chk4Reader *chunk.Chunk)
 	return err
 }
 
-func (t *mergeJoinInnerTable) rowsWithSameKey() ([]chunk.Row, error) {
-	lastResultIdx := len(t.resultQueue) - 1
-	t.resourceQueue = append(t.resourceQueue, t.resultQueue[0:lastResultIdx]...)
-	t.resultQueue = t.resultQueue[lastResultIdx:]
+func (t *mergeJoinInnerTable) selectRow(row chunk.Row) {
+	t.curSel = append(t.curSel, row.Idx())
+}
```

Comment:
How about removing this method since it just has one line? 👀

Reply:
I suppose it is more semantic and less confusing to have such a method, I prefer to keep.